barretenberg/cpp/src/barretenberg/honk/proof_system/ultra_verifier.cpp

```diff
@@ -157,16 +158,65 @@ template <typename Flavor> bool UltraVerifier_<Flavor>::verify_proof(const plonk
     // Produce a Gemini claim consisting of:
     // - d+1 commitments [Fold_{r}^(0)], [Fold_{-r}^(0)], and [Fold^(l)], l = 1:d-1
     // - d+1 evaluations a_0_pos, and a_l, l = 0:d-1
-    auto gemini_claim = Gemini::reduce_verification(multivariate_challenge,
-                                                    batched_evaluation,
-                                                    batched_commitment_unshifted,
-                                                    batched_commitment_to_be_shifted,
-                                                    transcript);
+    auto univariate_opening_claims = Gemini::reduce_verification(multivariate_challenge,
+                                                                 batched_evaluation,
+                                                                 batched_commitment_unshifted,
+                                                                 batched_commitment_to_be_shifted,
+                                                                 transcript);
+
+    // Perform ECC op queue transcript aggregation protocol
+    if constexpr (IsGoblinFlavor<Flavor>) {
+        // Receive commitments [t_i^{shift}], [T_{i-1}], and [T_i]
+        std::array<Commitment, Flavor::NUM_WIRES> prev_agg_op_queue_commitments;
+        std::array<Commitment, Flavor::NUM_WIRES> shifted_op_wire_commitments;
+        std::array<Commitment, Flavor::NUM_WIRES> agg_op_queue_commitments;
+        for (size_t idx = 0; idx < Flavor::NUM_WIRES; ++idx) {
+            prev_agg_op_queue_commitments[idx] =
+                transcript.template receive_from_prover<Commitment>("PREV_AGG_OP_QUEUE_" + std::to_string(idx + 1));
+            shifted_op_wire_commitments[idx] =
+                transcript.template receive_from_prover<Commitment>("SHIFTED_OP_WIRE_" + std::to_string(idx + 1));
+            agg_op_queue_commitments[idx] =
+                transcript.template receive_from_prover<Commitment>("AGG_OP_QUEUE_" + std::to_string(idx + 1));
+        }
+
+        // Receive transcript poly evaluations
+        FF kappa = transcript.get_challenge("kappa");
```

Comment:
Why not use rho from gemini? You have to open the t_shift on it anyway

Reply:
See related comment above